### DISCUSSION TITLE
Fixtures with zero density? Purely for collision?

### DISCUSSION THREAD

**Discussion by LordXerus** [dyn4j/dyn4j#225] [Q&A] [answered]
Is it possible to simulate fixtures with zero density? I'm using this to detect if an object is within a Camera's field of view.

Since the application is purely visual, I don't want any additional mass affecting the Physics simulation.

I know that zero-density fixtures aren't allowed because they can cause problems in simulation. But is it possible to accomplish this pure collision detection in parallel with the physics simulation? Or should I look for another library?

**Comment by wnbittle** [comment]
Are you asking how you can do static queries on the world?  You can do this in a number of ways, but there are two easy approaches.  Use the `World.detect(...)` methods which allow you to specify an AABB or arbitrary shape, or you can use the `World.raycast(...)` methods to detect anything that intersects a ray.

For example:
```java
World w = //
List<DetectResult<Body, BodyFixture>> results = w.detect(new AABB(5, 2));
for (DetectResult<Body, BodyFixture> result : results) {
    // ... do something with the results
}
```

**Reply by LordXerus** [reply]
I see. That's definitely helpful, but not quite what the question is asking.

I'm building a simulation with physics and graphics rendering (but since dyn4j is very complete it just seems like I'm adding graphics to dyn4j). I was wondering if I could use dyn4j to detect when the visible part of Bodies enter and leave the field of view. 

I want to 'attach' some massless 'fixtures' onto those bodies so I can detect when they enter or leave another shape(representing the field of view). They should be massless because they represent a something purely visual that should not impact the physics simulation in any way. However, I saw in the JavaDocs that zero-density fixtures are not allowed.

How should I approach this?

**Comment by wnbittle** [comment, marked as answer]
I see, I think what you are looking for are "sensor" fixtures, but these fixture are included in the total mass and shape of the body.  They don't participate in collision resolution, but will generate collision events.  An easy solution might be to just give them a small enough density that it doesn't effect the simulation.

It would be a pretty simple change to allow zero density though
